### ISSUE TITLE
Reset calculator when you select a different period

### ISSUE THREAD

**Issue by Borges-Porteous** [gc-proto/covid-19-guidance#62]
Didn't see this in the other issues, so adding one in.

If you fill out the calculator and then go change the period, the calculator still has the old amounts. Is there a way to just refresh the form when they select that other period in step 1?

**Comment by Christopher-O** [comment]
@Michel-Tremblay I have a feeling this would be a fairly easy onchange to remove all values from the inputs and reset selects and checkboxes within the form (something that mimics the reset button on a form).  Correct me if I'm wrong.   
And building on that, there was a desire to add reset buttons to each section (I don't know how indepth we should make that).

**Comment by Christopher-O** [comment]
@Borges-Porteous I wonder though if having the entire form reset when changing the period would be the expected behaviour?  

If someone spent an hour plunking in data, only to realize....oops, wrong period and then lost all their data, wouldn't that be a problem?  I'm wondering if instead we focus on using a reset button for the form, and if needed, once someone chooses a period, it turns to "readonly" and the only way to go back to default is by resetting.  At least then, the user is making the concious choice to reset the page and knows that they shouldn't plunk in data or else they will lose it.

**Comment by Borges-Porteous** [comment]
Yeah, a reset button would make sense to me - so they would have to hit 'reset' and then change the period? 

The reset resets the form and then the change would be as per usual.

**Comment by Christopher-O** [comment]
@Borges-Porteous correct.  If the major issue for this is the user changing the period then I see two options

- have a statement immediately below the dropdown for the period selection that says changing the period would reset the form,  however the risk is someone not seeing the statement

- auto change the selection to a readonly, and add a statement when that happens to say if they wish to change the form, they will need to reset the page.  At least that way, it gives them a chance to collect the data if they have already put it in, or not start putting in data at all

The first option would probably be the easiest to impliment, the second option would be a better failsafe for the user.  Honestly I'm good with either idea.

**Comment by Michel-Tremblay** [comment]
Has there been a decision on this? Which option are we going with?

**Comment by Borges-Porteous** [comment]
@Michel-Tremblay Hope this helps summarize:

1. Give an option to clear the form for step 2a that removes all field values
2. Issue #64  - but make step 1 unlock when they clear form
3. At the employee panel level, add a 'remove employee' button that deletes the panel and clears the numbers.

**Comment by Michel-Tremblay** [comment]
> At the employee panel level, add a 'remove employee' button that deletes the panel and clears the numbers.

will we need something to tell users they will need to resubmit the data to 2b if they remove/update employees after submitting the first time?

**Comment by Christopher-O** [comment]
@Michel-Tremblay That's a great point.  however the same would be said if they hit the submit and then said "oops I forgot employee X"...they would still have to resubmit the data then.  So if we do add that statement it would have to appear if they submit once and they either add or remove a new employee.

**Comment by Christopher-O** [comment]
These are all good points that we should gather for future updates.  Sigh...if only we had more time.

**Comment by Borges-Porteous** [comment]
> These are all good points that we should gather for future updates. Sigh...if only we had more time.

This. It's nice to have, not critical.
